homeassistant/components/asterisk_mbox.py

```diff
@@ -0,0 +1,192 @@
+"""Support for Asterisk Voicemail interface."""
+
+import asyncio
+import logging
+from contextlib import suppress
+
+import async_timeout
+import voluptuous as vol
+
+
+from aiohttp import web
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers import discovery
+from homeassistant.const import (CONF_HOST,
+                                 CONF_PORT, CONF_PASSWORD)
+
+from homeassistant.const import (HTTP_BAD_REQUEST,
+                                 HTTP_NOT_FOUND)
+
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import (async_dispatcher_connect,
+                                              async_dispatcher_send)
+
+from homeassistant.components.frontend import register_built_in_panel
+from homeassistant.components.http import HomeAssistantView
+
+REQUIREMENTS = ['asterisk_mbox==0.4.0']
+DEPENDENCIES = ['http']
+
+CONTENT_TYPE_MPEG = 'audio/mpeg'
+SIGNAL_MESSAGE_UPDATE = 'asterisk_mbox.message_updated'
+SIGNAL_MESSAGE_REQUEST = 'asterisk_mbox.message_request'
+
+DOMAIN = 'asterisk_mbox'
+ASTERISK = None
+
+_LOGGER = logging.getLogger(__name__)
+
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_PORT): int,
+        vol.Required(CONF_PASSWORD): cv.string,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up for the Asterisk Voicemail box."""
+    # pylint: disable=global-statement, import-error
+    global ASTERISK
```

Comment:
In a different review, I was asked not to use hass.data[] to store data (or at least that was my interpretation of the request), so I did my best to avoid them here.